tests/test_select.py

```diff
@@ -85,7 +86,7 @@ def test_select_default(
         pytest.param({"missing": "???"}, "missing", id="missing"),
     ],
 )
-def test_select_default_throw_on_missing(
+def test_select_default_throw_on_missing_dict(
```

Comment:
I feel like we discussed it but somehow this is still wrong.
this function is testing both dicts and lists.

Reply:
This is a side effect of how I had initially rebased this PR on top of a WIP version of #564 (where we discussed this point). It has been "auto-fixed" by the most recent rebase.

Reply:
Resolving since this is not part of this PR anymore